File: recursion_with_array.cpp
#include<iostream>
#include<vector>
using namespace std;

void display(vector<int> & arr,int vidx)
{
 if(vidx==arr.size())
 {return;
  }   
 cout<<arr[vidx]<<"  ";
 display(arr,vidx+1);   
}

bool find(vector<int> & arr,int vidx,int data)
{
 if(vidx==arr.size())
 {return 0 ;
  }  

 if(arr[vidx]==data)
 {return true;
  }

 find(arr,vidx+1,data);   
}

int maxm(vector<int> & arr,int vidx)
{
 if(arr[vidx]==arr.size())
 {return arr[vidx];
 }

 int res=maxm(arr,vidx+1);
}

void solve()
{
 vector<int> arr={10,6,8,10,4,5,5,6,8,-3,2,12,8,3};
 display(arr,0);
 
 cout<<endl;
 int data;

 cout<<"enter data you want to find";
 cin>>data;
 cout<<find(arr,0,data);

 cout<<endl;
 cout<<maxm(arr,0);

}

int main(int args,char** argv)
{
 solve();   
 
 return(0);

}

File: recursion_with_array.java
import java.util.Scanner;

public class recursion_with_array


{
 public static Scanner scn=new Scanner(System.in);

 public static void input(int arr[])
  {for(int i=0;i<arr.length;i++)
   arr[i]=scn.nextInt();
  } 


 public static void main(String[] args)
 {
  System.out.println("enter number of elements you want to enter");
  int n=scn.nextInt();

  int[] arr=new int [n];
  
  System.out.println("enter array elements");
  input(arr);

  System.out.println("enter number you want to find");
  int data=scn.nextInt();
 
  all_index(arr,0,data,a);

 }

 public static int[] all_index(int[] arr,int vidx,int data,int  a)
 {   
  if(vidx==arr.length)
  {
   int[] baseArray=new int [size];
   return baseArray;   
  }
  
  if(arr[vidx]==data)
  a++;

  int[] res = all_index(arr,vidx+1,data,a);

  if(arr[vidx]==data)
  {
   res[size-1]=vidx;
  }

  return res;

 }

}